Angle hms construct UnitsError

mhvk opened the pull request:
This way to create an Angle used to work with Astropy 1.1 (taking the tuple to mean hms format) but now errors with Astropy 1.2.dev15520 :
```python
>>> from astropy.coordinates import Angle
>>> angle = Angle((1, 2, 3), unit='hour')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/deil/code/astropy/astropy/coordinates/angles.py", line 94, in __new__
    angle = cls._tuple_to_float(angle, unit)
  File "/Users/deil/code/astropy/astropy/coordinates/angles.py", line 141, in _tuple_to_float
    .format(angle, unit))
astropy.units.core.UnitsError: Can not parse '(1, 2, 3)' as unit 'hour'
```
These work with the dev version:
```python
>>> Angle((1, 2, 3), unit='hourangle')
<Angle 1.0341666666666667 hourangle>
>>> Angle('1 2 3 hour')
<Angle 1.0341666666666667 hourangle>
>>> Angle((1, 2, 3), unit='deg')
<Angle 1.0341666666666667 deg>
```

Looks like all of those treat the tuple as dms:
```
>>> 1 + 2/60 + 3/60/60
1.0341666666666667
```
According to the Angle docs, it's possible to create from an hms tuple (although it isn't said what unit one should pass to do that):
http://astropy.readthedocs.io/en/latest/api/astropy.coordinates.Angle.html#astropy.coordinates.Angle

Assuming dms for unit='hourangle'` or 'hour' in the unit string is a bug, no?

I didn't check if this was introduced in #4970 or where the change in behaviour originated.

cc @mhvk @eteq

mhvk commented:
OK, turned out my changes had moved the conversion of strings to units to far back for non-Quantity input. Fix is now attached, as well as a regression test.

cdeil commented:
@mhvk - Thank you!

If this is accepted as a regression bug, shouldn't it go in v1.2?

Could you please add an hms tuple example [here](https://github.com/astropy/astropy/blob/f9c24683151d619aa132d028e23b860d534ccdfa/astropy/coordinates/angles.py#L53)? (at least that was the place where I checked in the docs what `unit` I should be passing for an hms tuple.

mhvk commented:
@cdeil - since the bug was introduced in #4970, which is for 1.3, this does not have to be back-ported to anything (thanks to you for finding it so quickly!). 

And, yes, good to put an example! I did use `'hourangle'` for it, as I think it is good to always think that one is giving an angle even if it is expressed as a time.

mhvk commented:
I realised my "regression test" actually did not test against the regression, using a unit instead of a string. This is now corrected; I also rebased it just to make it one commit.

@astrofrog - I think this is ready to go in once tests pass.

mhvk commented:
Merging since quite trivial...
